pkg/manager/leaderelection.go

```diff
@@ -33,3 +50,33 @@ func (k *KubeMacPoolManager) markPodAsLeader() error {
 	log.Info("marked this manager as leader for webhook", "podName", k.podName)
 	return nil
 }
+
+// By setting this status to true in all pods, we declare the kubemacpool as ready and allow the webhooks to start running.
+func (k *KubeMacPoolManager) setLeadershipConditions(status corev1.ConditionStatus) error {
+	podList := corev1.PodList{}
+	err := k.mgr.GetClient().List(context.TODO(), &podList, &client.ListOptions{Namespace: k.podNamespace})
+	if err != nil {
+		return err
+	}
+	for _, pod := range podList.Items {
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			podKey := types.NamespacedName{Namespace: k.podNamespace, Name: pod.Name}
+			err := k.mgr.GetClient().Get(context.TODO(), podKey, &pod)
+			if err != nil {
+				return err
+			}
+
+			pod.Status.Conditions = append(pod.Status.Conditions, corev1.PodCondition{Type: "kubemacpool.io/leader-ready", Status: status, LastProbeTime: metav1.Time{}})
```

Comment:
Let export `kubemacpool.io/leader-ready` to a variable or to a file like names.go